webui/react/src/components/kit/Input.tsx

```diff
@@ -1,16 +1,134 @@
-import { Input as AntdInput, InputProps, InputRef } from 'antd';
-import React, { forwardRef, ForwardRefExoticComponent, RefAttributes } from 'react';
+import {
+  Input as AntdInput,
+  InputProps as AntdInputProps,
+  Form,
+  FormItemProps,
+  InputRef,
+} from 'antd';
+import { PasswordProps as AntdPasswordProps } from 'antd/lib/input/Password';
+import { TextAreaProps as AntdTextAreaProps, TextAreaRef } from 'antd/lib/input/TextArea';
+import React, { forwardRef, ForwardRefExoticComponent, ReactNode, RefAttributes } from 'react';
 
-const Input: Input = forwardRef<InputRef, InputProps>((props: InputProps, ref) => {
-  return <AntdInput ref={ref} {...props} />;
-}) as Input;
+type Rules = FormItemProps['rules']; // https://github.com/ant-design/ant-design/issues/39466
+type LabelCol = {
+  span: number;
+};
+type TriggerEvent = 'onChange' | 'onSubmit';
+
+export interface WrapperProps {
+  children?: ReactNode;
+  label?: string;
+  labelCol?: LabelCol; // https://ant.design/components/grid#col
+  max?: number;
+  maxMessage?: string;
+  name?: string;
+  noForm?: boolean; // if not wrapped in an antd <Form> component
+  required?: boolean;
+  requiredMessage?: string;
+  rules?: Rules; // https://ant.design/components/form#rule
+  validateMessage?: string;
+  validateStatus?: '' | 'success' | 'warning' | 'error' | 'validating' | undefined;
+  validateTrigger?: TriggerEvent[];
+}
+
+export const FormItemWrapper: React.FC<WrapperProps> = ({
+  children,
+  label,
+  labelCol = { span: 24 },
+  name,
+  rules = [],
+  required,
+  requiredMessage,
+  max,
+  maxMessage,
+  validateMessage,
+  validateTrigger,
+  validateStatus,
+}: WrapperProps) => {
+  if (required) rules.push({ message: requiredMessage || `${label} required`, required: true });
+  if (max) rules.push({ max, message: maxMessage || `${label} cannot exceed ${max} characters` });
+
+  return (
+    <Form.Item
+      help={validateMessage}
+      label={label}
+      labelCol={labelCol}
+      name={name}
+      required={required}
+      rules={rules}
+      validateStatus={validateStatus}
+      validateTrigger={validateTrigger}>
+      {children}
+    </Form.Item>
+  );
+};
+
+type WrappedInputProps = AntdInputProps & WrapperProps;
+const Input: Input = forwardRef<InputRef, WrappedInputProps>(
+  ({ noForm, ...props }: WrappedInputProps, ref) => {
+    if (noForm) {
```

Comment:
Why is the component emitting its own form?

Reply:
the thinking was that we may want to allow an `Input` that's not in a `Form` to still have validation.

but rethinking it now, doesn't seem like we would want to use the same error handling, etc. for the `Input`s not in a `Form`, anyway.

I'll go ahead and remove the `noForm` prop usage, and perhaps just indicate in the DesignKit page that "required" and "invalid" states only apply when the component is used within a `Form`

Reply:
the other thing is that we just use the antd `Form` component without our own UI kit version. so I guess my initial thinking was to allow for this to be more independent from antd, as well.

maybe it makes sense to just add a `Form` component to the UI kit?

Reply:
If we are going to have these fields outside of a form then they should function outside of the form, not silently create their own.

You can add <Form> to the UI kit if you like. Also if we need non-form usecases it's fine to have a separate Form Item Wrapper in the UI kit.